app/Main.hs

```diff
@@ -486,10 +486,12 @@ readKubernetesJwt =
 -- | Authenticate using GitHub auth, see https://www.vaultproject.io/docs/auth/github.
 requestGitHubVaultToken :: Context -> Text -> IO (Either VaultError ClientToken)
 requestGitHubVaultToken context ghtoken = let
+  backendName = if null oBackendName then "github" else oBackendName
+        where oBackendName = getOptionsValue oVaultAuthBackend (cCliOptions context)
```

Comment:
```suggestion
  backendName = case oVaultAuthBackend (cCliOptions context) of
    Just name -> name
    Nothing -> "github"
```

If we change the default to value to `Nothing` as @Riscky suggested, to properly signal "not provided", instead of abusing empty string as a sentinel value, then here we can pick the backend name like this. [`getOptionsValue` would crash on the `Nothing`](https://github.com/channable/vaultenv/blob/eae5b8fa6b72c14134253062a13044f4f39497f4/src/Config.hs#L198-L200). Also below for the Kubernetes case.

It can also be written more concisely as 
```haskell
  backendName = fromMaybe "github" $ oVaultAuthBackend $ cCliOptions context
```
Personally I find the explicit pattern matching clearer.